src/main/webapp/script.js

```diff
@@ -373,8 +381,116 @@ function addInputDestinationLocationAutofill() {
   locationAutocomplete.addListener('place_changed', () => {
     inputDestination.classList.add('is-valid');
     checkNextButton();
+
+    // Remove any current elements, then get and add the suggested locations.
+    removeSuggestedLocations();
+    const radius = 50000;
+    let location = new google.maps.LatLng(locationAutocomplete.getPlace().geometry.location.lat(), 
+      locationAutocomplete.getPlace().geometry.location.lng());
+    getAndAddSuggestedLocations(location, radius);
   });
 }
 
+// Get suggested locations based on a central location (latitude, longitude) and
+// radius (meters). All suggested locations are of type "tourist attraction".
+// Then, call addSuggestedLocations function.
+function getAndAddSuggestedLocations(centralLocation, radius) {
+  let googlePlacesObject = new google.maps.places.PlacesService(document.createElement('div'));
+
+  let placesRequest = {
+    location: centralLocation,
+    radius: radius,
+    type: 'tourist_attraction'
+  };
+
+  googlePlacesObject.nearbySearch(placesRequest, (placesList, placesServiceStatus, _) => {
+    if (placesServiceStatus === google.maps.places.PlacesServiceStatus.OK) {
+      addSuggestedLocations(placesList);
+    }
+  });
+}
+
+// Removes all of the current suggested locations by setting innerHTML to empty.
+function removeSuggestedLocations() {
+  const suggestedLocationBlock = document.getElementById('suggested-location-block');
+  suggestedLocationBlock.innerHTML = '';
+}
+
+// Add suggested locations for POIs after user has submitted initial "Start 
+// Trip" form details (name of trip, location, and date).
+function addSuggestedLocations(suggestedLocations) {
+  const suggestedLocationBlock = document.getElementById('suggested-location-block');
+
+  // Add all of the suggested locations to the page.
+  suggestedLocations.forEach((location) => {
+    // If photo is present, get the photo source; if not, use placeholder.
+    let photoSrc;
+    if (location.photos !== undefined) {
+      photoSrc = location.photos[0].getUrl();
+    } else {
+      photoSrc = 'images/placeholder_image.png';
+    }
+    
+    const suggestedLocationWidget = buildSuggestedLocationWidget(location.name,
+      location.vicinity, photoSrc);
+    suggestedLocationBlock.appendChild(suggestedLocationWidget);
+  });
+}
+
+// Builds and returns an HTML widget of a suggested location.
+function buildSuggestedLocationWidget(name, vicinity, photoSrc) {
+  // The container that holds the full card.
+  const cardContainer = document.createElement('div');
+  cardContainer.className = 'card';
+  cardContainer.style = 'width: 15rem';
+
+  // Add the photo of this location.
+  const photoElement = document.createElement('img');
+  photoElement.className = 'card-img-top';
```

Comment:
No action required for this CL, but what this method does would be made much easier by a templating library. Barring any objection from the VSE's, I think it might be time to incorporate handlebars.js. I'd be happy to help with the ramp-up on that.

Reply:
Ok, that sounds good -- last time I messaged the VSE chat with this message I didn't get a response, but I just sent them another message.